src/game_api/script.cpp

```diff
@@ -867,6 +872,39 @@ SpelunkyScript::ScriptImpl::ScriptImpl(std::string script, std::string file, Sou
         return sol::nullopt;
     };
 
+    /// Loads save.dat from the script's pack. If save.dat does not exist, you will get an empty string.
+    lua["load_data"] = [this]() -> std::string
+    {
+        std::ifstream datafile;
+        datafile.open(script_folder / "save.dat");
+        if (datafile.good()) 
+        {
+            std::string string;
+
+            datafile.seekg(0, std::ios::end);   
+            string.reserve(datafile.tellg());
+            datafile.seekg(0, std::ios::beg);
+
+            string.assign((std::istreambuf_iterator<char>(datafile)),
```

Comment:
I've had some bad experiences with this API for reading whole files in. Just to keep in mind that if there are ever issues mentioned we might want to move over to the C API (the only C API that I personally prefer to the C++ API 😄)
No need to act on this now.

Reply:
Alright.